File: openfaas/resolvers/subjects/handler.js
'use strict'
const fs = require('fs')

let db = JSON.parse(fs.readFileSync(`${__dirname}/common/database.json`));

module.exports = async ({body: {parent, args, ctx, info, headers}}, context) => {
  console.log("ARGUMENTS FOR 'subjects'", args, ctx, info, headers, ctx)
  
  let data = null
  if (parent && parent.id) {
    data = db.studySubjects[parent.id]
  }
  
  return context
    .status(200)
    .headers({"content-type": "application/json"})
    .succeed(JSON.stringify(data || null),)
}

File: openfaas/resolvers/sites/handler.py
import json
import pathlib

database_file = pathlib.Path(__file__).parent / "common/database.json"
db = json.load(database_file.open())

def handle(req):
    body = req.get_json()
    parent_id = body.get("parent", {}).get("id", None)

    study_subjects = db['studySubjects']

    data = None
    if parent_id:
        data = study_subjects.get(parent_id, {}).get("site", {})

    return data, 200, {"Content-Type":"application/json"}
